src/test/java/io/cdap/plugin/ariba/source/AribaInputFormatTest.java

```diff
@@ -0,0 +1,243 @@
+                                                           "08ee0299-4849-42a4-8464-3abed75fc74e",
+                                                           "c3B5wvrEsjKucFGlGhKSWUDqDRGE2Wds",
+                                                           "xryi0757SU8pEyk7ePc7grc7vgDXdz8O",
+                                                           "2022-01-28T10:05:02Z", "2022-01-31T10:05:02Z");
+    aribaInputFormat = new AribaInputFormat();
+    ObjectMapper objectMapper = new ObjectMapper();
+    JsonNode crateJob = objectMapper.readTree(createJobResponse);
+    JsonNode jobData = objectMapper.readTree(createJobResponse1);
+    new Expectations(AribaServices.class) {
+      {
+        aribaServices.createJob(pluginConfig, anyString);
+        result = crateJob;
+        minTimes = 0;
+
+        aribaServices.fetchJobStatus(anyString, anyString);
+        result = new AribaException("limit exceed", 429);
+        minTimes = 0;
+
+        aribaServices.getAccessToken();
+        result = "token";
+        minTimes = 0;
+      }
+    };
+
+    try {
+      aribaInputFormat.createJob(pluginConfig, aribaServices, false, null);
+    } catch (IOException io) {
+      Assert.assertEquals(io.getMessage(), "limit exceed");
+    }
+  }
+
+  @Test
+  public void testCreateJobThrowError3() throws IOException, AribaException, InterruptedException {
+    AribaPluginConfig pluginConfig = new AribaPluginConfig("unit-test-ref-name",
+                                                           "https://openapi.au.cloud.ariba.com",
+                                                           "prod",
+                                                           "CloudsufiDSAPP-T",
+                                                           "SourcingProjectFactSystemView",
+                                                           "08ee0299-4849-42a4-8464-3abed75fc74e",
+                                                           "c3B5wvrEsjKucFGlGhKSWUDqDRGE2Wds",
+                                                           "xryi0757SU8pEyk7ePc7grc7vgDXdz8O",
+                                                           "2022-01-28T10:05:02Z", "2022-01-31T10:05:02Z");
+    aribaInputFormat = new AribaInputFormat();
+    ObjectMapper objectMapper = new ObjectMapper();
+    JsonNode crateJob = objectMapper.readTree(createJobResponse);
+    JsonNode jobData = objectMapper.readTree(createJobResponse1);
+    new Expectations(AribaServices.class) {
+      {
+        aribaServices.createJob(pluginConfig, anyString);
+        result = new InterruptedException();
+        minTimes = 0;
+
+        aribaServices.fetchJobStatus(anyString, anyString);
+        result = jobData;
+        minTimes = 0;
+
+        aribaServices.getAccessToken();
+        result = "token";
+        minTimes = 0;
+      }
+    };
+    try {
+      aribaInputFormat.createJob(pluginConfig, aribaServices, false, "kk");
+    } catch (IOException io) {
+      Assert.assertNull(io.getMessage());
```

Comment:
add comment explaining why we expect an error to be thrown

Reply:
Fixed